keps/0005-cluster-resources-for-crds.md

```diff
@@ -85,25 +88,49 @@ should probably be created
 
 OperatorVersions require the existance of CRDs that are not controlled by Kudo (e.g. ETCD Operator) and require those to be installed when FV is enabled.
 
-### Implementation Details/Notes/Constraints [optional]
+### Implementation Details/Notes/Constraints
 
+The installation of an Operator Version can run a plan specified by the Operator Version spec, when the `installPlan` value is specified in `operator.yaml`.  
+The value shall match a defined `plan` in the `plans` spec in the OperatorVersion. As with the Instance Status properties, the OperatorVersion will contain
 
+```golang
+	PlanStatus       map[string]PlanStatus `json:"planStatus,omitempty"`
+	AggregatedStatus AggregatedStatus      `json:"aggregatedStatus,omitempty"`
+```
 
-### Risks and Mitigations
+to contain the status information on the execution of the `installPlan`. The runtime logic for executing the plan on OperatorVersion install will be the same core logic that is used in the Instance plan execution.
+
+#### Instance Status update
+
+If Instances of an OperatorVersion are created before the Install plan of the OperatorVersion is successful, the Instance's will remian in a `PENDING` status until the OperatorVersion is successful.
+
+#### Object Owners
 
+The objects created by the `installPlan` may exist on the cluster before the OperatorVersion is installed. This may be a result of another OperatorVersion requiring the same objects, or the objects being managed externally to KUDO. If so, the OperatorVersion will be appended to the `ownerReferences`. If the object does not exist, then KUDO will create the object with the OperatorVersion as the `ownerReference`, and additionally add an annotation to the object that it was created by KUDO: `kudo.dev/managed: true`. This will be used during cleanup to identify whether the object was created by KUDO or not.
 
+If the object already exists, but does not match the spec provided by the `installPlan`, the `installPlan` will fatally error and the OperatorVersion's status will be set to `FatalError`. We will not support updating/changing existing objects as part of an install since this may change how components not managed by this OperatorVersion would function.
+
+#### Uninstalls and Object Cleanup
+
+When an OperatorVersion is uninstalled, KUDO will remove the OperatorVersion from the `ownerReference`. If there are no more `ownerReferences`, it will look for the `kudo.dev/managed: true` annotation to identify if this object is no longer needed on the cluster. If that annotation is there, it will remove the object. If it is not there, then the object must have existed before installing any OperatorVersions and is managed by a component outside of KUDO, and will remain on the cluster.
```

Comment:
One potential problem that I see here is that we might remove a CRD that is used by somebody else. We handle the case where the CRD is already present before KUDO tries to install it, by adding KUDO to the ownerReferences.

Can we safely assume that any other user of a certain CRD would add itself to the CRD's ownerReferences as well if it uses it? I.e., KUDO installs CRD X, then another software A tries to install CRD X as well, detects that it is already installed and doesn't add itself to the ownerReferences. If software A then creates CRs of type X and we remove the CRD X, all CustomResources get deleted.

We might be able to check all instances of CR X for "kudo.dev/managed: true" annotations, although that could be a lot at some point.

Not sure if this is a realistic scenario, or too far fetched, but deletion of something that's not created by KUDO would be *really* bad.